src/components/molecules/Visualizer/Plugin/api.ts

```diff
@@ -188,6 +193,74 @@ export function exposed({
             return commonReearth.plugins.instances;
           },
         },
+        clientStorage: {
+          get getAsync() {
+            return (key: string) => {
+              const promise = clientStorage.getAsync(
+                (plugin?.extensionType === "widget"
+                  ? widget?.()?.id
+                  : plugin?.extensionType === "block"
+                  ? block?.()?.id
+                  : "") ?? "",
```

Comment:
I don't think you need the final `?? ""` since you are already falling back to `""` if not a widget or block

Reply:
That is for the case that `widget?.()?.id` or `block?.()?.id` is `undefined`.